modules/sdk-lib-mpc/src/tss/ecdsa/paillierBlumProof.ts

```diff
@@ -0,0 +1,137 @@
+import { createHmac } from 'crypto';
+import { bitLength, randBits, isProbablyPrime } from 'bigint-crypto-utils';
+import { modInv, modPow } from 'bigint-mod-arith';
+import { bigIntFromBufferBE, bigIntToBufferBE } from '../../util';
+import { DeserializedPaillierBlumProof } from './types';
+
+// Security parameter.
+const m = 80;
+
+/**
+ * Generate psuedo-random quadratic residue for (N, w, i).
+ * @param N - the prime number to verify is a product of two large primes.
+ * @param w - a random number with the same bitLength as N, that satisfies the Jacobi of w is -1 wrt N.
+ * @returns {bigint[]} - set of challenges for N
+ */
+function generateY(N, w): bigint[] {
+  const NBuf = bigIntToBufferBE(N);
+  const wBuf = bigIntToBufferBE(w, NBuf.length);
+  return Array(m)
+    .fill(null)
+    .map((_, i) => {
+      const h = bigIntFromBufferBE(
+        createHmac('sha256', Buffer.from([i]))
```

Comment:
i think according to https://www.zkdocs.com/docs/zkdocs/protocol-primitives/nums/, we need to use an XOF hash function to produce a digest of the bitlength of N.

Reply:
maybe something like SHAKE-256

Reply:
sha3-XOF  (shake-256 mode) is typically used for that, especially when doing RSA keys

Reply:
Updated to use sha256 and separators for each field. However, TypeScript complains about the required parameters to createHash. See code comment.